### PR TITLE
Update cloc

### LINKED ISSUE #6161: cloc does not work on Windows?
`npm run cloc` reports "The command line is too long."  Perhaps because of the number of source files and their length?

Works on Mac and Linux (via CI).

I am running Windows 7 still.  If this works in Windows 10, let's just close.

### PR DESCRIPTION
**Pull request by hpinkos**
Fixes #6161

- Get cloc from npm (current version is 1.74)
- Don't cloc Specs/Data

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@hpinkos, thanks for the pull request! Maintainers, we have a signed CLA from @hpinkos, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: gulpfile.js [hunk 1]
```diff
@@ -137,17 +137,14 @@ gulp.task('requirejs', function(done) {
...
-    var cloc_definitions = path.join('Tools', 'cloc-1.60', 'cloc_definitions');
+    var clocPath = path.join('node_modules', 'cloc', 'lib', 'cloc');
 
     //Run cloc on primary Source files only
     var source = new Promise(function(resolve, reject) {
-        var glsl = globby.sync(['Source/Shaders/*.glsl', 'Source/Shaders/**/*.glsl']).join(' ');
-
-        cmdLine = 'perl ' + clocPath + ' --quiet --progress-rate=0 --read-lang-def=' + cloc_definitions +
-                  ' Source/main.js Source/Core/ Source/DataSources/ Source/Renderer/ Source/Scene/ Source/Widgets/ Source/Workers/ ' + glsl;
+        cmdLine = 'perl ' + clocPath + ' --quiet --progress-rate=0' +
```
Comment (ggetz) [inline gulpfile.js]: Just to clarify, we run this after `clean` so this only counts the `.glsl` shaders and not the generated `.js` ones, correct? I don't think blowing away the built files will be an expected consequence of running this task, can we just exclude the built files?

Also is it still necessary to run this with perl? I think you should be able to call the executable from `node_modules` directly.

File: gulpfile.js [hunk 2]
```diff
@@ -163,7 +160,8 @@ gulp.task('cloc', ['build'], function() {
     //If running cloc on source succeeded, also run it on the tests.
     return source.then(function() {
         return new Promise(function(resolve, reject) {
-            cmdLine = 'perl ' + clocPath + ' --quiet --progress-rate=0 --read-lang-def=' + cloc_definitions + ' Specs/';
+            cmdLine = 'perl ' + clocPath + ' --quiet --progress-rate=0' +
```
Comment (ggetz) [inline gulpfile.js]: Do we want to exclude .eslintrc.json and the karma configurations?

**Comment by mramato** [comment]
@hpinkos any idea why there is a GLSL and XML file showing up in this branch for the Specs but not in master?  Otherwise the numbers look spot on.

File: gulpfile.js (hunk 1 above)
Comment (hpinkos) [reply gulpfile.js]: cloc isn't an executable, it's a perl script

yes, clean gets rid of the `js` files so we're only counting the `glsl` shaders

this is the last step in the travis build, so there shouldn't be any problems running `clean`

**Comment by hpinkos** [comment]
@mramato yep, it's because we were clocing the `Specs/Data` folder

File: gulpfile.js (hunk 1 above)
Comment (mramato) [reply gulpfile.js]: @ggetz see #6161, we can't easily ignore them without larger structure changes.

File: gulpfile.js (hunk 2 above)
Comment (mramato) [reply gulpfile.js]: I wouldn't, since it's all "code".

**Comment by mramato** [comment]
@hpinkos master doesn't have them, they are new in this branch.

File: gulpfile.js (hunk 2 above)
Comment (hpinkos) [reply gulpfile.js]: We were counting them before so I think it's okay to leave as is

**Comment by mramato** [comment]
Nevermind, I totally got my tavs confused.  This branch is correct and master isn't.  Thanks @hpinkos.  @ggetz can merge when she's happy.

File: gulpfile.js (hunk 1 above)
Comment (ggetz) [reply gulpfile.js]: OK with me then.

**Pull request merged**